ql/src/java/org/apache/hadoop/hive/ql/plan/ConditionalResolverMergeFiles.java

```diff
@@ -208,14 +235,17 @@ public List<Task<?>> getTasks(HiveConf conf, Object objCtx) {
           } else {
             // static partition and list bucketing
             generateActualTasks(conf, resTsks, trgtSize, avgConditionSize, mvTask, mrTask,
-                mrAndMvTask, dirPath, inpFs, ctx, work, lbLevel, manifestFilePresent);
+                mrAndMvTask, dirPath, inpFs, ctx, work, lbLevel, manifestFilePresent, storageHandler);
           }
         }
+      } else if (useCustomStorageHandler) {
+        generateActualTasks(conf, resTsks, trgtSize, avgConditionSize, mvTask, mrTask,
+                mrAndMvTask, dirPath, inpFs, ctx, work, 0, false, storageHandler);
       } else {
         Utilities.FILE_OP_LOGGER.info("Resolver returning movetask for " + dirPath);
         resTsks.add(mvTask);
       }
-    } catch (IOException e) {
+    } catch (ClassNotFoundException | HiveException | IOException e) {
```

Comment:
should we handle ClassNotFoundException | HiveException in generateActualTasks?

Reply:
Added another `catch` clause with these 2 exceptions.